Samples, Resailers and updates

Delmael opened the pull request:
Hello hello 😄 

As I have resumed the work on the smartbox package, I was able to update the hass component.
You can see here the changelog for the pypi package : https://github.com/Delmael/smartbox/releases/tag/v2.1.0-beta.2

## What's Changed

* Every interaction are now by default async and we use the hass-websession. This is an unvisible update but needed for beeing more compliant with hass requirements.
* Resailers : 
    * I've simplified the config flow form. You do not need anymore to know the api name and the basic auth. I get this information from the smartbox package. If the resailer is not in the list we will have to update the smartbox package with a new release. Start from now, you only need email and password for configuration flow.
    * When you are on the device interface,  you can directly go to the configuration web page of this device
    * We also can, with an option config flow, display the logo of the resailers instead of hass logo for entities
* Samples : the smartbox API give you the consumption and temperature history of the device through samples.
    * There is a new power sensor which get this information. The energy dashboard is now correctly working !
    * The first time the config entry is set, i try to get the last 3 years of consumption and inject it in statistics. Next after, every hour i can the last 24 hours of consumption and update it. You can reset/active/off this features throw the options config.
* Entities : I've marked some of them as configuration or diagnostic entities.

⚠️  _**This integration required Home assistant >= 2015 and python 3.13**_ ⚠️

ajtudela commented:
Awesome impovmentes

> As I have resumed the work on the smartbox package, I was able to update the hass component. You can see here the changelog for the pypi package : https://github.com/Delmael/smartbox/releases/tag/v2.1.0-beta.2
Please also make a PR with these changes to the main repo to have one with the main development.

>   * I've simplified the config flow form. You do not need anymore to know the api name and the basic auth. I get this information from the smartbox package. If the resailer is not in the list we will have to update the smartbox package with a new release. Start from now, you only need email and password for configuration flow.
That's a big improvment!

>   * There is a new power sensor which get this information. The energy dashboard is now correctly working !
How does it work? Is it reset every 24 hours so that the energy dashboard shows the correct information?

I'll test it tomorrow, but LGTM.

P.D.: I need to give you maintainer permissions for the repo :)

Delmael commented:
> Awesome impovmentes
> 
> > As I have resumed the work on the smartbox package, I was able to update the hass component. You can see here the changelog for the pypi package : https://github.com/Delmael/smartbox/releases/tag/v2.1.0-beta.2
> > Please also make a PR with these changes to the main repo to have one with the main development.
> 
> > * I've simplified the config flow form. You do not need anymore to know the api name and the basic auth. I get this information from the smartbox package. If the resailer is not in the list we will have to update the smartbox package with a new release. Start from now, you only need email and password for configuration flow.
> >   That's a big improvment!
> 
> > * There is a new power sensor which get this information. The energy dashboard is now correctly working !
> >   How does it work? Is it reset every 24 hours so that the energy dashboard shows the correct information?
> 
> I'll test it tomorrow, but LGTM.
> 
> P.D.: I need to give you maintainer permissions for the repo :)

I can make a PR to update atjudela/smartbox but I have configured all the pypi, links, issue templates and other stuff with my Delmael/smartbox repo.....

Regarding the consumption history, I'v added into the README a section to explain a little bit how it works. It's maybe not enough so let me know what I need to add 😄 

I've removed the ignore brands information into the hacs workflow as we now have a logo.

ajtudela commented:
> I can make a PR to update atjudela/smartbox but I have configured all the pypi, links, issue templates and other stuff with my Delmael/smartbox repo.....
That would be weird, having the repos under differents namespaces but... OK.

Delmael commented:
> > I can make a PR to update atjudela/smartbox but I have configured all the pypi, links, issue templates and other stuff with my Delmael/smartbox repo.....
> 
> That would be weird, having the repos under differents namespaces but... OK.

@ajtudela I can make all the modification and PR to merge into your atjudela/smartbox, but i will have to give me the maintainer role on your repo :)

ajtudela commented:
> @ajtudela I can make all the modification and PR to merge into your atjudela/smartbox, but i will have to give me the maintainer role on your repo :)

Done!

Delmael commented:
> > @ajtudela I can make all the modification and PR to merge into your atjudela/smartbox, but i will have to give me the maintainer role on your repo :)
> 
> Done!

Hum I saw, and accept the invitation, but I can't access to the repo settings :s

ajtudela commented:
> > > @ajtudela I can make all the modification and PR to merge into your atjudela/smartbox, but i will have to give me the maintainer role on your repo :)
> > 
> > Done!
> 
> Hum I saw, and accept the invitation, but I can't access to the repo settings :s

Mm, 5hat's a GitHub thing, only the owner of the repo can access the settings.

Delmael commented:
> > > > @ajtudela I can make all the modification and PR to merge into your atjudela/smartbox, but i will have to give me the maintainer role on your repo :)
> > > 
> > > 
> > > Done!
> > 
> > 
> > Hum I saw, and accept the invitation, but I can't access to the repo settings :s
> 
> Mm, 5hat's a GitHub thing, only the owner of the repo can access the settings.

Oh. That's not really practical....
Do u want me to send what need to be configured ?

ajtudela commented:
> > > > > @ajtudela I can make all the modification and PR to merge into your atjudela/smartbox, but i will have to give me the maintainer role on your repo :)
> > > > 
> > > > 
> > > > Done!
> > > 
> > > 
> > > Hum I saw, and accept the invitation, but I can't access to the repo settings :s
> > 
> > Mm, 5hat's a GitHub thing, only the owner of the repo can access the settings.
> 
> Oh. That's not really practical....
> Do u want me to send what need to be configured ?

Yes, send me an email, please.
